controllers/external/util.go

```diff
@@ -152,6 +152,19 @@ func FailuresFrom(obj *unstructured.Unstructured) (string, string, error) {
 	return failureReason, failureMessage, nil
 }
 
+// FailureDomain returns the FailureDomain from the external object spec.
+func FailureDomain(obj *unstructured.Unstructured) *string {
+	result, found, err := unstructured.NestedFieldNoCopy(obj.Object, "spec", "failureDomain")
+	if err != nil || !found {
+		return nil
+	}
+	failureDomain, ok := result.(*string)
```

Comment:
I don't _think_ an unstructured object would have a `*string` - when it decodes json, it should be coming in as a `string`.